subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/JavaRecompilationSpecProvider.java

```diff
@@ -19,34 +19,41 @@
 import com.google.common.base.MoreObjects;
 import com.google.common.collect.ImmutableSet;
 import org.gradle.api.file.FileTree;
+import org.gradle.api.file.FileType;
 import org.gradle.api.internal.file.FileOperations;
 import org.gradle.api.internal.file.FileTreeInternal;
 import org.gradle.api.internal.tasks.compile.JavaCompileSpec;
 import org.gradle.api.internal.tasks.compile.incremental.processing.GeneratedResource;
 import org.gradle.api.tasks.WorkResult;
-import org.gradle.api.tasks.incremental.IncrementalTaskInputs;
 import org.gradle.api.tasks.util.PatternSet;
 import org.gradle.internal.Factory;
+import org.gradle.work.ChangeType;
+import org.gradle.work.FileChange;
 
 import java.io.File;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.EnumMap;
 import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.gradle.internal.FileUtils.hasExtension;
 
 public class JavaRecompilationSpecProvider extends AbstractRecompilationSpecProvider {
-    private final IncrementalTaskInputs inputs;
+    private final boolean incremental;
+    private final Supplier<Iterable<FileChange>> sourceFileChanges;
     private final JavaConventionalSourceFileClassNameConverter sourceFileClassNameConverter;
 
-    public JavaRecompilationSpecProvider(FileOperations fileOperations, FileTreeInternal sources, IncrementalTaskInputs inputs, CompilationSourceDirs sourceDirs) {
+    public JavaRecompilationSpecProvider(FileOperations fileOperations, FileTreeInternal sources, boolean incremental, Supplier<Iterable<FileChange>> sourceFileChanges, CompilationSourceDirs sourceDirs) {
```

Comment:
If I pass in the input changes object, I also need to pass in the sources object, since this is required when querying for input changes for the sources.

Reply:
True, but you wouldn't need to a) pass in the `boolean` and b) wrap the thing in `Supplier`. I don't feel strongly either way.

Reply:
I am passing `Iterable` now, no need for a `Supplier`.